model/versioning/Versioned.php

```diff
@@ -1196,12 +1196,96 @@ public function doPublish() {
 		$owner = $this->owner;
 		$owner->invokeWithExtensions('onBeforePublish');
 		$owner->write();
-		$owner->publish("Stage", "Live");
+		$owner->publish(static::DRAFT_STAGE, static::LIVE_STAGE);
 		$owner->invokeWithExtensions('onAfterPublish');
 		return true;
 	}
 
+	/**
+	 * Trigger publishing of owned objects
+	 */
+	public function onAfterPublish() {
+		$owner = $this->owner;
+
+		// Publish owned objects
+		$published = new ArrayList();
+		foreach ($owner->findOwned(false, $published) as $object) {
```

Comment:
How does the extended class know it needs to implement this?

Can we either check for this method & no-op if missing, or, better,  have an interface, like `Owner`, which specifies this method and we check for `is_a('Owner')` and no-op if not.